Entities/Characters/Builder/BuilderLogic.as

```diff
@@ -142,26 +142,45 @@ bool RecdHitCommand(CBlob@ this, CBitStream@ params)
 
 	if (blobID == 0)
 	{
-		// block
 		CMap@ map = getMap();
 		if (map !is null)
 		{
 			uint16 type = map.getTile(tilepos).type;
 			if (!inNoBuildZone(map, tilepos, type))
 			{
-				if (getNet().isServer())
-				{
-					map.server_DestroyTile(tilepos, 1.0f, this);
 
-					Material::fromTile(this, type, 1.0f);
+				CBlob@[] blobs_here;
+				map.getBlobsAtPosition(tilepos + Vec2f(1, 1), blobs_here);
+
+				bool no_dmg = false;
+
+				// dont dmg backwall if there's a door here
+				for (int i=0; i < blobs_here.size(); ++i)
+				{
+					CBlob@ current_blob = blobs_here[i];
+					if (current_blob !is null && (current_blob.hasTag("door") || current_blob.getName() == "bridge" || current_blob.getName() == "wooden_platform"))
+					{
+						no_dmg = true;
+						break;
+					}
 				}
 
-				if (getNet().isClient())
+				if (!no_dmg)
 				{
-					if (map.isTileBedrock(type))
+					if (isServer())
+					{
+						map.server_DestroyTile(tilepos, 1.0f, this);
+
+						Material::fromTile(this, type, 1.0f);
+					}
+
+					if (isClient())
 					{
-						this.getSprite().PlaySound("/metal_stone.ogg");
-						sparks(tilepos, attackVel.Angle(), 1.0f);
+						if (map.isTileBedrock(type))
```

Comment:
You can check this before `isServer()` on line 151/152 and if this is true you don't need to execute the entire `isServer()` on line 152-175 so it can start with `else if`.

Reply:
resolved if I understood properly